src/models/vac.ts

```diff
@@ -1,21 +1,44 @@
-import { parseISO, isBefore, isAfter } from 'date-fns'
+import dayjs from "dayjs";
+import utc from "dayjs/plugin/utc";
+import customParseFormat from "dayjs/plugin/customParseFormat";
+import { parseISO, isBefore, isAfter } from 'date-fns';
 import { Vaccination, VaccinationRecord, VaccineRecord } from "../types";
 import * as sddInformations from "../static/SDD.mapping.json";
 
+dayjs.locale("en-sg");
+dayjs.extend(utc);
+dayjs.extend(customParseFormat);
+
 /**
  * Helper function to generate Eu Vaccination records.
  * @param vaccinationRecord
  * @returns
  */
 export const genVaccinationRecord = (vaccinationRecord: VaccinationRecord, expiryDateTime: string, issuerName: string, reference: string): VaccineRecord[] => {
+  const validatedVaccinationRecords = (vaccinationRecord.vaccinations).map((vaccination) => {
+    const validateVaccinationDateTime = vaccination.vaccinationDateTime.split('T')[0]; // take only date value if ISO-8601 format
+    if (
+      !dayjs(
+        validateVaccinationDateTime,
+        ["YYYY-MM-DD"],
+        true
+      ).isValid()
+    ) {
+      throw new Error(
+        `Invalid vaccinationDateTime (${vaccination.vaccinationDateTime}). Should be YYYY-MM-DD format`
```

Comment:
Note : EU required date format is like this;
<img width="623" alt="Screenshot 2022-05-15 at 10 12 59 PM" src="https://user-images.githubusercontent.com/10098074/168477200-04df1767-691f-48d3-b2c8-16639aad333e.png">